Arrowgene.Ddon.Server/Scripting/interfaces/GameLogicSetting.cs

```diff
@@ -0,0 +1,630 @@
+using Arrowgene.Ddon.Server.Scripting.utils;
+using Arrowgene.Ddon.Shared.Model;
+using System.Collections.Generic;
+
+namespace Arrowgene.Ddon.Server.Scripting.interfaces
+{
+    public class GameLogicSetting
+    {
+        private ScriptableSettings SettingsData { get; set; }
+        public GameLogicSetting(ScriptableSettings settingsData)
+        {
+            SettingsData = settingsData;
+        }
+
+        private T GetSetting<T>(string key)
+        {
+            return SettingsData.Get<T>("GameLogicSettings", key);
+        }
+
+        public T Get<T>(string scriptName, string key)
```

Comment:
The getters in this file are simply for compatibility with what currently exists. I've used a different pattern for some of the new code I wrote for this PR as well.